playwright-e2e/utils/test-utils.ts

```diff
@@ -40,22 +41,31 @@ export async function waitForNoSpinner(page: Page, opts: { timeout?: number } =
 
 export async function waitForResponse(page: Page, { uri, status = 200, timeout, messageBody }: WaitForResponse): Promise<Response> {
   let response: any;
-  if (messageBody) {
-    response = await page.waitForResponse((response: Response) => response.url().includes(uri) &&
-    messageBody.every(async message => (await response.text()).includes(message as string)), { timeout });
-  } else {
-    response = await page.waitForResponse((response: Response) => response.url().includes(uri), { timeout });
+  try {
+    if (messageBody) {
+      response = await page.waitForResponse((response: Response) => response.url().includes(uri) &&
+      messageBody.every(async message => (await response.text()).includes(message as string)), { timeout });
+    } else {
+      response = await page.waitForResponse((response: Response) => response.url().includes(uri), { timeout });
+    }
+    await response.finished();
+  } catch (err) {
+    if (err instanceof errors.TimeoutError) {
+      throw new Error(`TimeoutError: waiting for URI: ${uri}: Timeout exceeded`);
```

Comment:
better error in test log.

Example

<img width="963" alt="Screenshot 2024-01-29 at 7 46 43 PM" src="https://github.com/broadinstitute/ddp-angular/assets/35533885/b660f61e-4152-4588-943e-3022467069a5">